packages/lesswrong/components/common/CloudinaryImage.jsx

```diff
@@ -0,0 +1,39 @@
+import { Components, registerComponent, getSetting } from 'meteor/vulcan:core';
+import React from 'react';
+
+function cloudinaryPropsToStr(props) {
+  let sb = [];
+  for(let k in props)
+    sb.push(k+'_'+props[k]);
+  return sb.join(",");
+}
+
+const CloudinaryImage = (props, context) => {
+  const cloudinaryCloudName = getSetting('cloudinary.cloudName', 'lesswrong-2-0')
+  
+  let cloudinaryProps = {
+    c: "fill",
+    dpr: "auto",
+    g: "custom",
+    q: "auto",
+  };
+  let imageStyle = {};
+  
+  if(props.width) {
+    cloudinaryProps.w = props.width;
+    imageStyle.width = props.width+"px";
+  }
+  if(props.height) {
+    cloudinaryProps.h = props.height;
+    imageStyle.height = props.height+"px";
+  }
+  
+  const imageUrl = `http://res.cloudinary.com/${cloudinaryCloudName}/image/upload/${cloudinaryPropsToStr(cloudinaryProps)}/${props.publicId}`;
+  
+  return <img
+    src={imageUrl}
+    style={imageStyle}
+  />
+};
+
+registerComponent('CloudinaryImage', CloudinaryImage);
```

Comment:
Consider exporting this using an es6 `export default CloudinaryImage` export instead of using registerComponent

Reply:
These aren't exactly substitutes. While we're not using it in this instance, the main function of `registerComponent` is to provide a hook to hang higher-order component wrappers like `withStyles` on. So if we switched from mostly using `registerComponent` and `Components.Foo` to mostly using `export default` and `Foo`, the HoCs would mean we only end up switching part way, which would be awkward.

Reply:
It seems fine to me to slowly migrate towards using "registerComponent" only where we need it for the HOCs, and using "export" in all other places.